alea/src/covariance.cpp

```diff
@@ -79,13 +82,32 @@ template <typename T, typename Str>
 void cov_acc<T,Str>::add(const computed<value_type> &source, size_t count)
 {
     internal::check_valid(*this);
-    source.add_to(sink<T>(current_.sum().data(), current_.size()));
+    source.add_to(view<T>(current_.sum().data(), current_.size()));
     current_.count() += count;
 
     if (current_.is_full())
         add_bundle();
 }
 
+template <typename T, typename Str>
+cov_acc<T,Str> &cov_acc<T,Str>::operator<<(const cov_result<T,Str> &other)
+{
+    internal::check_valid(*this);
+    if (size() != other.size())
+        throw size_mismatch();
+
+    // NOTE partial sums are unchanged
+    // HACK we need this for "outwardly constant" manipulation
```

Comment:
This is ugly indeed... Can you think of a better approach?

Reply:
> This is ugly indeed... Can you think of a better approach?

Yes, but I would like to wait until I have refactored these classes into a base class and a set of accumulator classes for different dimensionality.  Otherwise I have to do the work twice. I don't think it is a big issue though.